### ISSUE TITLE
Add function signature to the sharpen function

### ISSUE THREAD

**Issue by nateisleychamp** [lovell/sharp#2561]
We call functions like resize and modulate with their parameters in an object, which results in calls like:
```javascript
sharp(input)
  .modulate({
    brightness: 0.5,
    saturation: 0.5,
    hue: 90
  });

sharp(input)
  .resize({ width: 100, height: 300 });
```

But this does not work:
```javascript
sharp(input).sharpen({ sigma: 2, flat: 2, jagged: 3 }
```
What are you trying to achieve?
Symmetry with how I call other functions

Have you searched for similar feature requests?
Yes. My search did not turn up any issues.

**Comment by lovell** [comment]
Happy to accept a PR, if you're able, that adds the ability to use this approach whilst maintaining the existing API.

**Comment by lovell** [comment]
The Object-based approach for setting `sharpen` parameters is now available as of v0.30.3. The previous approach still works and is now deprecated.

https://sharp.pixelplumbing.com/api-operation#sharpen
